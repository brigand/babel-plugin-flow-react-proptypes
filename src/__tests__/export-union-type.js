const babel = require('babel-core');
const content = `
// @flow
export type A = "option1" | "option2";
`;

it('export-union-type', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
