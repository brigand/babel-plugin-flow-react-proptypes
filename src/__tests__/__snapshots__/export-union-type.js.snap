// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-union-type 1`] = `
"\\"use strict\\";

var babelPluginFlowReactPropTypes_proptype_A = require(\\"prop-types\\").oneOf([\\"option1\\", \\"option2\\"]);

if (typeof exports !== \\"undefined\\") Object.defineProperty(exports, \\"babelPluginFlowReactPropTypes_proptype_A\\", {
  value: babelPluginFlowReactPropTypes_proptype_A,
  configurable: true
});"
`;
